opennms-doc/guide-admin/src/asciidoc/text/kafka-producer/kafka-producer.adoc

```diff
@@ -0,0 +1,219 @@
+
+==== Alarms
+
+The Kafka Producer listens for changes made to the current set of alarms and forwards the resulting alarms to a Kafka topic.
+The records are keyed by alarm reduction key and contain a GPB encoded model of the alarm.
+When an alarm is deleted, a `null` value is sent with the corresponding reduction key.
+Publishing records in this fashion allows the topic to be used as a link:https://docs.confluent.io/current/streams/concepts.html#ktable[KTable].
+The Kafka Producer will also perform periodic synchronization tasks to ensure that the contents of the Kafka topic reflect the current state of alarms in the _{opennms-product-name}_ database.
+
+By default, all alarms (and subsequent updates) are forwarded to a topic named `alarms`.
+
+The name of the topic used can be configured, and an optional filtering expression can be set to help control which alarms are sent to the topic.
+
+==== Nodes
+
+If an event or alarm being forwarded reference a node, then the corresponding node is also forwarded.
+The records are keyed by "node criteria" (see bellow) and contain a GPB encoded model of the alarm.
+A caching mechanism is in place to help avoid forwarding nodes that have been successfully forwarded, and have not changed since.
+
+The name of the topic used can be configured.
+
+IMPORTANT: The node topic is not intended to include all of the nodes in the system, it only includes records for nodes that relate to events or alarms that have been forwarded.
+
+===== Node Criteria
+
+The node criteria is a string representation of the unique identifier for a given node.
+If the node is associated with a foreign source (fs)  and foreign id (fid), the node criteria resulting node criteria will be the name of the foreign source, followed by a colon (:) and then the foreign id i.e. (fs:fid).
+If the node is not associated with both a foreign source and foreign id, then the node id (database id) will be used.
+
+=== Enabling the Kafka Producer
+
+The Kafka Producer is not enabled by default and can be enabled as follows.
+
+First, login to the Karaf shell of your _{opennms-product-name}_ instance and configure the Kafka client settings to point to your Kafka broker.
+See link:https://kafka.apache.org/10/documentation.html#producerconfigs[Producer Configs] for a complete list of available options.
+
+[source]
+----
+$ ssh -p 8101 admin@localhost
+...
+admin@opennms()> config:edit org.opennms.features.kafka.producer.client
+admin@opennms()> config:property-set bootstrap.servers 127.0.0.1:9092
+admin@opennms()> config:update
+----
+
+Next, install the `opennms-kafka-producer` feature from that same shell using:
+[source]
+----
+admin@opennms()> feature:install opennms-kafka-producer
+----
+
+In order to ensure that the feature continues to be installed as subsequent restarts, add `opennms-kafka-producer` to the `featuresBoot` property in the `${OPENNMS_HOME}/etc/org.apache.karaf.features.cfg`
```

Comment:
Is there a way to add this to a dedicated file which can be easily updated? As described here, this makes updating and maintaining opennms harder and error prune.